### DISCUSSION TITLE
items_list null in FattureInCloud\Model\Receipt

### DISCUSSION THREAD

**Discussion by gyosko** [fattureincloud/api#260] [Q&A] [answered]
Ciao,

avrei bisogno di accedere alle righe registrate all'interno di un `FattureInCloud\Model\Receipt`, ma la `items_list` è sempre null.

I Receipts in questo momento vengono presi tramite questa chiamata:

```
$receipts = $receipts_api->listReceipts(
	$company_id,
	null,
	null,
	1,
	500,
	null,
	$q
)->getData();
```

E vengono ciclati così:

```
foreach ($receipts as $r) {...}
```

La chiamata a `$items = $r->getItemsList()` però ritorna sempre null, anche se sono presenti delle righe e il totale è diverso da zero.

Sarebbe possibile recuperare le righe in altro modo?

Anche un `json_encode($r)` mostra un oggetto con `items_list:null`, ma dalla dashboard di fatture in cloud le righe vengono visualizzate correttamente.

**Comment by emajo** [comment, marked as answer]
Ciao @gyosko per far restituire il campo `items_list` popolato è necessario passare `fieldset = detailed`, in questo caso usando l'sdk: 
```
$receipts = $receipts_api->listReceipts(
	$company_id,
	null,
	'detailed',
	1,
	500,
	null,
	$q
)->getData();
```

Grazie,
Emanuele.
